src/connect.ts

```diff
@@ -31,7 +31,7 @@ export async function calculateActualConfig(config: FileSystemConfig): Promise<F
   config.host = replaceVariables(config.host);
   const port = replaceVariables((config.port || '') + '');
   if (port) config.port = Number(port);
-  config.agent = replaceVariables(config.agent);
+  config.agent = replaceVariables(config.agent) || process.env.SSH_AUTH_SOCK;
```

Comment:
I haven't thought of using this variable, but it's a good idea. If you want, you can make this a separate PR, or let me commit it myself.